packages/flame/test/experimental/layout_component_test.dart

```diff
@@ -0,0 +1,147 @@
+import 'package:flame/components.dart';
+import 'package:flame/experimental.dart';
+import 'package:flame/game.dart';
+import 'package:flame_test/flame_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  late Component component;
+  late Vector2 gameSize;
+
+  late PositionComponent firstComponent;
+  late PositionComponent secondComponent;
+  late PositionComponent thirdComponent;
+  late double totalSizeOfComponents;
+
+  void _initChildren() {
+    firstComponent = PositionComponent(
+      size: Vector2(50, 50),
+    );
+
+    secondComponent = PositionComponent(
+      size: Vector2(100, 100),
+    );
+
+    thirdComponent = PositionComponent(
+      size: Vector2(150, 150),
+    );
+  }
+
+  for (var k = 0; k <= 1; k++) {
+    group(k == 0 ? 'RowComponent' : 'ColumnComponent', () {
```

Comment:
would this be better as:
```
for (final name in ['RowComponent', 'ColumnComponent']) {
```
or, maybe even better, extract the block to a function and call it twice?

```
group('RowComponent', testLayoutComponent(RowComponent))
group('ColumnComponent', testLayoutComponent(ColumnComponent))

```